### ISSUE TITLE
Incorrect B006 when aliasing Mapping

### ISSUE THREAD

**Issue by cdleonard** [astral-sh/ruff#28735]
### Summary

Example Code:

```py
from typing import Mapping


def myfun(d: Mapping[str, str] = {}) -> str:
    result = ""
    for k, v in d.items():
        result += f"{k}={v}\n"
    return result


CustomMapping = Mapping[str, str]


def myfun2(d: CustomMapping = {}) -> str:
    result = ""
    for k, v in d.items():
        result += f"{k}={v}\n"
    return result
```

First function does not trigger B006 but the second one does

### Version

0.16.8

**Comment by MichaReiser** [comment]
We can fix this when the type alias is defined in the same file, but not for type aliases imported from other files (requires full type inference)
